packages/angular_devkit/build_optimizer/src/build-optimizer/webpack-plugin.ts

```diff
@@ -0,0 +1,35 @@
+
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { Compiler } from 'webpack';  // tslint:disable-line:no-implicit-dependencies
+
+export class BuildOptimizerWebpackPlugin {
+  apply(compiler: Compiler) {
+    compiler.hooks.normalModuleFactory.tap("BuildOptimizerWebpackPlugin", nmf => {
+      nmf.hooks.module.tap("BuildOptimizerWebpackPlugin", (module, data) => {
+        const resolveData = data.resourceResolveData;
+        if (
+          resolveData &&
+          resolveData.descriptionFileData &&
+          resolveData.relativePath
+        ) {
+          // Only TS packages should use Build Optimizer.
+          const typings = resolveData.descriptionFileData.typings;
+          // Note: a TS package might not have defined typings but still use .d.ts files next to
```

Comment:
Maybe an additional note that this will also enable build optimizer for JS origin packages that happen to have a typings file?

Reply:
Yes, that's another limitation. Added.